### PR TITLE
Car reorder and improvement

### LINKED ISSUE #285: Reorder car order
Look at attached file. 

[LODO - Biler.xlsx](https://github.com/dodoas/lodo/files/542727/LODO.-.Biler.xlsx)

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #285

### CODE REVIEW

File: modules/car/model/car.class.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?
+class car {
+  // function that replaces Active field in queries.
+  // $car_id can be a fixed number, or an sql statement
+  // examples:
+  //   1. $car_id = 4;
+  //   2. $car_id = "car.CarID"; This is in examples like: $all_active_cars = "SELECT * FROM car WHERE ". car::car_active_sql("car.CarID") ."=1;";
+  //   3. $car_id = "(SELECT CarID FROM voucher WHERE voucher.VoucherID = 123)"; This sends a specific CarID for which you want to see if it is active.
+  public static function car_active_sql($car_id, $reference_date = false) {
```
Comment (Baxxx) [inline modules/car/model/car.class.php:9]: Discuss about reference dates (use login form date/voucher date?)

File: modules/fakturabank/model/fakturabank.class.php [hunk 2]
```diff
@@ -897,7 +897,8 @@ private function validate_incoming($invoicesO) {
                 # validate invoice lines
                 foreach($InvoiceO->InvoiceLine as &$line) {
                   if ($line->Item->AdditionalItemProperty->Name == 'Car') {
-                    $query = "select * from car where CarCode='" . $line->Item->AdditionalItemProperty->Value . "' and Active=1";
+                    includelogic("car/car");
+                    $query = "select * from car where CarCode='" . $line->Item->AdditionalItemProperty->Value . "' and ". car::car_active_sql("car.CarID") ."=1";
```
Comment (Baxxx) [inline modules/fakturabank/model/fakturabank.class.php]: Check with issue date as reference date

File: modules/bank/view/record.inc [hunk 3]
```diff
@@ -155,6 +155,12 @@ function do_bank_update($AccountID, $Period) {
 
                     //#Speedier to have it here? Could we throw away the inner loop?
                     if($table_name == 'accountline') { //#We are just working on accountline?
+                        $CarID = $DataH[$server_name][$database_name][$table_name][$pk_name]['CarID'];
+                        $car_active_ref_date = $Period."-".$DataH[$server_name][$database_name][$table_name][$pk_name]['Day'];
+                        if ($CarID != "" && !car::is_active($CarID, $car_active_ref_date)) {
+                            $DataH[$server_name][$database_name][$table_name][$pk_name]['CarID'] = 0;
```
Comment (Baxxx) [inline modules/bank/view/record.inc:163]: Show a warning if this happens.

**Comment by sls-lnluksa** [comment]
First round review: Nikola Lukovic (sls-lnluksa)
Final review with: @stoivo

**Pull request merged**